modules/phase_field/src/materials/KKSPhaseConcentrationMaterial.C

```diff
@@ -46,17 +48,37 @@ KKSPhaseConcentrationMaterial::KKSPhaseConcentrationMaterial(const InputParamete
     _prop_ci(_num_c * 2),
     _ci_old(_num_c * 2),
     _ci_IC(getParam<std::vector<Real>>("ci_IC")),
+
     _f1(getMaterial("F1_material")),
     _f2(getMaterial("F2_material")),
+    _prop_fi(2),
+    _fi_copy(2),
+    _fi_copy_name(getParam<std::vector<MaterialPropertyName>>("Fi_copy")),
+
     _Fi_names(getParam<std::vector<MaterialPropertyName>>("Fi_names")),
     _first_dFi(2),
     _second_dFi(2),
     _iter(declareProperty<Real>("nested_iterations")),
     _abs_tol(getParam<Real>("absolute_tolerance")),
     _rel_tol(getParam<Real>("relative_tolerance")),
-    _nested_solve(NestedSolve(parameters))
+    _nested_solve(NestedSolve(parameters)),
+
+    _prop_dFidci(2),
+    _dFidci_names(2)
 
 {
+  _prop_dFidci[0] = &declareProperty<Real>(_dFidci_names[0]);
+  _dFidci_names[0] = DerivativeMaterialPropertyNameInterface::derivativePropertyNameFirst(
+      _Fi_names[0], _ci_names[0]);
+  // _dFidci[0] = &getMaterialPropertyDerivative<Real>(_Fi_names[0], _ci_names[0]);
```

Comment:
@dschwen Here I want to compute the free energy derivatives and generate the dFi/dci names automatically. We talked about do not directly use the DerivativeMaterialPropertyNameInterface, but use getMaterialPropertyDerivative instead. I am unsure how that works to get the derivative names...